### ISSUE TITLE
parsing integers partial success

### ISSUE THREAD

**Issue by flexoron** [infradig/trealla#245]
?-  length(N,1?).
Error: missing operand to '?', line 0, '(null)'

?-  length(N,1?1).
N = [_4].

just for fun
?-  length(N,1 :).
Error: missing operand to ':', line 0, '(null)'
?-  length(N,1 :-).
Error: missing operand to ':-', line 0, '(null)'
?-  length(N,1 :-?-).
N = [_4].

**Comment by flexoron** [comment]
$ ./tpl 
Trealla Prolog (c) Infradig 2020-2021, v1.13.10
?- length(N,2\this_is_a_comment).
N = [_4,_5].
?-

**Comment by UWN** [comment]
> interesting?

How do you discover these?

**Comment by flexoron** [comment]
by habit because whitespace parsing is somehow special:
https://en.wikipedia.org/wiki/Whitespace_(programming_language)

$ ./tpl 
Trealla Prolog (c) Infradig 2020-2021, v1.13.22
?- length(N,2\ :-?-). % a space between \ and :
N = [_4,_5].
?- length(N,2\ : -?-). % and between : and -
N = [_4,_5].
?- length(N,2\ : - ?-). % and between - and ?
N = [_4,_5].
?- length(N,2\ : - ? -). and between ? and -  ... aha parser does something after the backslash but what and why?
Error: missing operand to '-', line 0, ''

ok...here is your operand:
?- length(N,2\ : - ? - N = [a,b,c]).
N = [_5,_6].

**Comment by UWN** [comment]
But how did you find the initial example? From there on, it's plausible.

The minimal example would have been `a\b.` So an exhaustive enumeration would have found it quickly, too. Trouble is, not even numbers are conforming...

**Comment by flexoron** [comment]
first test case " length(N,1?1)" from above partially fixed.

$ ./tpl 
Trealla Prolog (c) Infradig 2020-2021, v1.13.28

?-  length(N,1?1).
Error: syntax error, near '?' operator expected '' 
false.

?-  length(N,1?-1).
N = [_4].
?-

**Comment by infradig** [comment]
Yes, you can do that with any pre/post-fix op.

**Comment by flexoron** [comment]
not sure whats right whats wrong.
here is another example:

$ ./tpl 
Trealla Prolog (c) Infradig 2020-2021, v1.13.28

?-  length(N,10).
N = [_4,_5,_6,_7,_8,_9,_10,_11,_12,_13].

?-  length(N,1 0).
N = [_4,_5,_6,_7,_8,_9,_10,_11,_12,_13].

?-  length(N,01).
N = [_4].

?-  length(N,0 1).
Error: syntax error, near '1' operator expected ''
?-

swipl:
?- length(N,0 1).
N = [_9276].

scryer and gprolog do not accept this kind of  digit grouping.

**Comment by infradig** [comment]
I don't really like space digit grouping, underscore I can go with.

**Comment by infradig** [comment]
In fact am going to remove it.

**Comment by UWN** [comment]
Since you permit `1_1`, consider to allow optional layout after the `_`. In this manner big integers can be printed and read from a fixed column format (like 80 chars).

```
        digit separator
           = underscore char (* 6.5.2 *),
             [ layout text sequence (* 6.4.1 *) ] ;
```

**Comment by infradig** [comment]
Pardon my ignorance, but what is layout here?

**Comment by UWN** [comment]
> Pardon my ignorance, but what is layout here?
layout characters like space, tab, newline and comments

**Comment by flexoron** [comment]
what kind of error is this?

$ ./tpl 
Trealla Prolog (c) Infradig 2020-2021, v1.13.31
?- X = 1, Y = X.
X = 1, Y = 1.

?- X = 1 (=), Y = X.
X = 1.

?- X = 1 (=); Y = X.
X = 1.

?- X = a (=); Y = X.
Error: syntax error, or operator expected, line 0: a, ''
?-

**Comment by flexoron** [comment]
$ ./tpl
Trealla Prolog (c) Infradig 2020-2021, v1.16.5
?- X is 1 + (3 is 1 + 1).
X = 3.
?-

**Comment by flexoron** [comment]
fixed with v1.20.11
